lib/billing_facade_client/cost_code_validator.rb

```diff
@@ -0,0 +1,27 @@
+module BillingFacadeClient
+  class CostCodeValidator < ActiveModel::Validator
+    SPLIT_CHARACTER = '-'
+
+    COST_CODE_REGEXP_STR = "\\AS[\\d]{4}(#{BillingFacadeClient::CostCodeValidator::SPLIT_CHARACTER}[0-9]{1,2}){0,1}\\z"
```

Comment:
I think {0, 1} could be replaced with a question mark to better show that that part is optional